### PR TITLE
Remove unnecessary casts

### PR DESCRIPTION
**Pull request by nschloe**
From https://github.com/scipy/scipy/pull/11147.

### CODE REVIEW

File: scipy/sparse/tests/test_base.py [hunk 1]
```diff
@@ -780,7 +780,7 @@ def check_setdiag(a, b, k):
             # correct length, and too short or too long vectors
             for r in [-1, len(np.diag(a, k)), 2, 30]:
                 if r < 0:
-                    v = int(np.random.randint(1, 20, size=1))
+                    v = np.random.randint(1, 20, size=1)[0]
```
Comment (ilayn) [inline scipy/sparse/tests/test_base.py]: If we are touching this already could you please use `np.random.choice(range(1, 20))`  which a bit more readable than the mysterious [0]?

**Review by ilayn — APPROVED** [review]
Apart from the single line I think this is good to go regardless of the casting changes.

File: scipy/sparse/tests/test_base.py (hunk 1 above)
Comment (nschloe) [reply scipy/sparse/tests/test_base.py]: Did that.

**Comment by ilayn** [comment]
Thanks @nschloe !

**Comment by mdhaber** [comment]
Thanks @nschloe ! If you'd like your name added to [THANKS.txt](https://github.com/scipy/scipy/blob/master/THANKS.txt), please make an entry [here](https://github.com/scipy/scipy/wiki/*THANKS.txt-additions-modifications-for-1.5.0*).

**Pull request merged**